commands/removerole.js

```diff
@@ -0,0 +1,142 @@
+        return message.channel.send(help).then(r => r.delete(30000))
+    }
+    if(!message.member.hasPermission("MANAGE_ROLES")){
+        message.reply("You don't have permission").then(async m=>{
+            await message.delete()
+            return m.delete(15000)
+        })
+    }
+    var message_id = args[0]
+    if (!args[1]) {
+        // remove the whole record that includes this message
+        let removeall = new Discord.RichEmbed()
+            .setColor("#FFA500")
+            .addField("Warning", "You didn't provide a role/emoji to remove, would you like to remove all the roles/emojis bounded to this message?")
+            .addField("Respone", "Type **Yes** to overwrite or anything else to Cancel")
+            .setDescription("*This command will be automatically cancelled in 60 seconds*")
+        msg = message.channel.send(removeall)
+        const filter = m => m.author.id === message.author.id
+        var response = await message.channel.awaitMessages(filter, { max: 1, time: 60000 }).then(async collected => {
+            if (!collected.first()) {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FF0000")
+                    .addField("Failture", "Command timed out")
+                ).then(m => m.delete(30000))
+                return false
+            }
+            if (collected.first().content.toLowerCase() == "yes") {
+                return true
+            } else {
+                msg.edit(new Discord.RichEmbed()
+                    .setColor("#FFFF00")
+                    .addField("Information", "Command Cancelled")
+                ).then(m => m.delete(30000))
+                return false
+            }
+        }).catch(r => { console.log(r) })
+        if (!response) {
+            return
+        }
+        reaction_message_model.deleteOne({
+            messageID: message_id
+        }, (err, rem) => {
+            if (err) { console.log(err); return }
+            if (!rem) {
+                let noq = new Discord.RichEmbed()
+                    .setColor("#F0FFF0")
+                    .addField("Notification", "No roles were found for this message in the database to remove.")
+                return message.channel.send(noq).then(r => r.delete(30000))
+            } else {
+                let done = new Discord.RichEmbed()
+                    .setColor("#00FFF0")
+                    .addField("Notification", `All roles/emojis on message with the id **${message_id}** were deleted from the database.`)
+                return message.channel.send(done).then(r => r.delete(30000))
+            }
+        })
+    } else {
+        // remove only the specified emoji or corresponding role
+        reaction_message_model.findOne({
+            messageID: message_id
+        }, async (err, rem) => {
```

Comment:
Unless necessary (i.e. to avoid rebinding of this) arrow functions are discouraged for anything more than a quick one-line filter or transform. Avoid arrow functions for anything more than a one-liner.
```suggestion
        }, async function(err, rem) {
```